src/Events/Calendar_Embeds/Admin/List_Page.php

```diff
@@ -0,0 +1,305 @@
+	}
+
+	/**
+	 * Removes the filters and actions hooks added by the controller.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function unregister(): void {
+		remove_action( 'admin_menu', [ $this, 'register_menu_item' ], 11 );
+		remove_action( 'adminmenu', [ $this, 'restore_menu_globals' ] );
+		remove_filter( 'submenu_file', [ $this, 'keep_parent_menu_open' ] );
+		remove_action( 'manage_' . Calendar_Embeds::POSTTYPE . '_posts_custom_column', [ $this, 'manage_column_content' ] );
+		remove_filter( 'manage_' . Calendar_Embeds::POSTTYPE . '_posts_columns', [ $this, 'manage_columns' ] );
+	}
+
+	/**
+	 * Customize columns for the table.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $columns The columns.
+	 *
+	 * @return array
+	 */
+	public function manage_columns( array $columns ): array {
+		$new_columns = [
+			'cb'               => $columns['cb'] ?? '<input type="checkbox" />',
+			'title'            => __( 'Calendar Embeds', 'the-events-calendar' ),
+			'event_categories' => __( 'Categories', 'the-events-calendar' ),
+			'event_tags'       => __( 'Tags', 'the-events-calendar' ),
+			'snippet'          => __( 'Embed Snippet', 'the-events-calendar' ),
+		];
+
+		/**
+		 * Filters the columns for the calendar embeds list table.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $new_columns The columns.
+		 *
+		 * @return array The filtered columns.
+		 */
+		return (array) apply_filters( 'tec_events_calendar_embeds_list_table_columns', $new_columns );
+	}
+
+	/**
+	 * Customize the content of the columns.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $column_name The name of the column.
+	 * @param int    $post_id     The post ID.
+	 *
+	 * @return void
+	 */
+	public function manage_column_content( $column_name, $post_id ): void {
+		switch ( $column_name ) {
+			case 'event_categories':
+				// Get events categores from post meta.
+				$categories = Calendar_Embeds::get_event_categories( $post_id );
+				if ( empty( $categories ) ) {
+					echo esc_html( __( 'All Categories', 'the-events-calendar' ) );
+					break;
+				}
+
+				$cat_markup = [];
+				foreach ( $categories as $category ) {
+					ob_start();
+					?>
+					<a href="<?php echo esc_url( get_edit_term_link( $category, TEC::TAXONOMY ) ); ?>">
+						<?php echo esc_html( trim( $category->name ) ); ?></a>
+					<?php
+					$cat_markup[] = ob_get_clean();
+				}
+
+				// phpcs:ignore StellarWP.XSS.EscapeOutput.OutputNotEscapedExpected, WordPress.Security.EscapeOutput.OutputNotEscaped, StellarWP.XSS.EscapeOutput.OutputNotEscaped
+				echo implode( ', ', array_map( 'trim', $cat_markup ) );
+				break;
+			case 'event_tags':
+				// Get events tags from post meta.
+				$tags = Calendar_Embeds::get_tags( $post_id );
+				if ( empty( $tags ) ) {
+					echo esc_html( __( 'All Tags', 'the-events-calendar' ) );
+					break;
+				}
+
+				$tag_markup = [];
+				foreach ( $tags as $tag ) {
+					ob_start();
+					?>
```

Comment:
A nice to have would be to move these cases into separate render methods.

Reply:
Done !